### PR TITLE
feat(application): implement newrelic_application resource

### PR DESCRIPTION
**Pull request by zlesnr**
resolves: #538

### CODE REVIEW

File: website/docs/r/application.html.markdown [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+layout: "newrelic"
+page_title: "New Relic: newrelic_application"
+sidebar_current: "docs-newrelic-resource-application"
+description: |-
+  Manage configuration for an existing application in New Relic.
+---
+
+# Resource: newrelic_application
+
+Use this resource to manage configuration for an application that already exists in New Relic.
```
Comment (jthurman42) [inline website/docs/r/application.html.markdown:11]: Should make a larger call-out that you can not Create a new application this way.

Comment (sanderblue) [reply website/docs/r/application.html.markdown:11]: Could use one of the markdown callouts that TF supports...

`~>` for a beige background callout
`->` for a blue background callout 

There's also one for green, but can't remember what it is off the top of head.

Comment (zlesnr) [reply website/docs/r/application.html.markdown:11]: Updated to include a brighter note.

**Comment by zlesnr** [comment]
Fixes were made here: https://github.com/terraform-providers/terraform-provider-newrelic/pull/559. Will cherry pick after merge.

File: website/docs/r/application.html.markdown [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The ID of the application.
+
+## Import
+
+Applications can be imported using notation `application_id`, e.g.
```
Comment (sanderblue) [inline website/docs/r/application.html.markdown:41]: 💭 Is `application_id` mentioned somewhere else? Thoughts on wording this something like "Applications can be imported using the application ID, e.g."

Comment (sanderblue) [reply website/docs/r/application.html.markdown:41]: Oh I bet this is a NerdGraph field

**Comment by jthurman42** [comment]
Changes went into master for the updated name, those need to be cherry picked back onto this release branch @zlesnr

**Pull request merged**